sql-parser/src/main/java/io/crate/sql/Identifiers.java

```diff
@@ -65,13 +65,8 @@ public static String quoteIfNeeded(String identifier) {
     }
 
     private static boolean areQuotesRequired(String identifier) {
-        for (int i = 0; i < identifier.length(); i++) {
-            int cp = identifier.codePointAt(i);
-            if (cp == '"' || cp == '[' || cp == ']' || cp == '(' || cp == ')' || Character.isUpperCase(cp)) {
-                return true;
-            }
-        }
-        return isKeyWord(identifier);
+        Matcher matcher = IDENTIFIER.matcher(identifier);
+        return isKeyWord(identifier) || !matcher.matches();
```

Comment:
```suggestion
        return isKeyWord(identifier) || !IDENTIFIER.matcher(identifier).matches();
```

To benefit from the short-cuircuiting of `||` and not run the matcher logic if `isKeyWord` is already true.

Reply:
oh, sure!